blocks/api/test/serializer.js

```diff
@@ -129,7 +117,7 @@ describe( 'block serializer', () => {
 					{ fruit: 'Bananas' }
 				);
 
-				expect( saved ).toBe( '<div>Bananas</div>' );
```

Comment:
Do you know why the className was missing here?

Reply:
>Do you know why the className was missing here?

Yep, alluded to in https://github.com/WordPress/gutenberg/pull/4786#issuecomment-363133291 with the "fixed test case" mention, but the issue was specifically in how `createElement` is treated differently from calling `save` as a function directly. It's the same reason why we couldn't just refactor to `createElement`:

![image](https://user-images.githubusercontent.com/1779930/35820112-b94e2ab2-0a72-11e8-9146-e26a8970a4d9.png)

...which is that, when cloning the element produced by `createElement`, it's the same as modifying props of what is passed into `save`, _not_ what is returned by `save`. The filtering expects to be applied on whatever is produced by `save`, e.g. the `div` here.

We don't really have a way to hook into the render behavior without some awful machinery to `render` into a temporary DOM, so instead we drop support for `save` as a component and just treat it as a function (i.e. it's still not a "stateless function component", just a function returning an element(s)).